src/components/application_manager/test/commands/mobile/subscribe_vehicle_data_request_test.cc

```diff
@@ -390,6 +403,82 @@ TEST_F(SubscribeVehicleDataRequestTest, OnEvent_AppAlreadySubscribed_Ignored) {
   command->on_event(event);
 }
 
+TEST_F(SubscribeVehicleDataRequestTest,
+       OnEvent_OtherAppIsSubscribedToIVI_SUCCESS) {
+  (*msg_)[am::strings::params][am::strings::connection_key] = kKey;
+  (*msg_)[am::strings::msg_params][kKeyName] = true;
+  CommandPtr command(CreateCommand<SubscribeVehicleDataRequest>(msg_));
+
+  ON_CALL(mock_app_manager_, application(kKey)).WillByDefault(Return(app_));
+  // Data is subscribed
+  EXPECT_CALL(*app_, IsSubscribedToIVI(data_type)).WillOnce(Return(false));
+
+  MockAppPtr other_app(CreateMockApp());
+  am::ApplicationSet app_set;
+  app_set.insert(other_app);
+  EXPECT_CALL(mock_app_manager_, applications())
+      .WillOnce(
+          Return(DataAccessor<am::ApplicationSet>(app_set, app_set_lock_)));
+  EXPECT_CALL(*other_app, IsSubscribedToIVI(data_type)).WillOnce(Return(true));
+
+  EXPECT_CALL(*app_, SubscribeToIVI(data_type)).WillOnce(Return(true));
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_result::SUCCESS), _));
+
+  command->Run();
+
+  Event event(Event::EventID::VehicleInfo_SubscribeVehicleData);
+  MessageSharedPtr event_msg(CreateMessage(smart_objects::SmartType_Map));
+  const hmi_apis::Common_Result::eType hmi_result_code =
+      hmi_apis::Common_Result::SUCCESS;
+  const mobile_apis::Result::eType mobile_result_code =
+      mobile_apis::Result::SUCCESS;
+  (*event_msg)[am::strings::params][am::hmi_response::code] = hmi_result_code;
+  (*event_msg)[am::strings::msg_params][kKeyName][am::strings::result_code] =
+      hmi_apis::Common_VehicleDataResultCode::VDRC_SUCCESS;
+  event.set_smart_object(*event_msg);
+
+  EXPECT_CALL(*application_manager::MockMessageHelper::message_helper_mock(),
+              HMIToMobileResult(hmi_result_code))
+      .WillOnce(Return(mobile_result_code));
+  const std::string info = "";
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(CheckResponseWithKeyParams(
+                              mobile_result_code,
+                              info,
+                              true,
+                              kKeyName,
+                              data_type,
+                              mobile_apis::VehicleDataResultCode::VDRC_SUCCESS),
+                          _));
+  EXPECT_CALL(*app_, UpdateHash());
+  command->on_event(event);
+}
+
+TEST_F(SubscribeVehicleDataRequestTest, Run_HasDisallowedParams_DISALLOWED) {
+  (*msg_)[am::strings::params][am::strings::connection_key] = kKey;
```

Comment:
@LevchenkoS I can see that you renamed kKey to kConnectionKey. Please use this rename, it is good.